erpnext/controllers/selling_controller.py

```diff
@@ -372,6 +372,9 @@ def set_po_nos(self):
 
 	def set_pos_for_sales_invoice(self):
 		po_nos = []
+		if self.po_no:
+			po_nos.append(self.po_no)
+			
```

Comment:
Instead of appending we could check if self. po_no exist in` def set_po_nos(self):` and if it exists we can skip calling the below functions which are overriding the values.

Reply:
The current implementation keeps both, the purchase orders from the user as well as system generated